docker/daemon/daemon_dest.go

```diff
@@ -248,15 +178,21 @@ func (d *daemonImageDestination) PutSignatures(signatures [][]byte) error {
 // - Uploaded data MAY be removed or MAY remain around if Close() is called without Commit() (i.e. rollback is allowed but not guaranteed)
 func (d *daemonImageDestination) Commit() error {
 	logrus.Debugf("docker-daemon: Closing tar stream")
-	if err := d.tar.Close(); err != nil {
-		return err
-	}
-	if err := d.writer.Close(); err != nil {
+	if err := d.img.Commit(); err != nil {
 		return err
 	}
 	d.committed = true // We may still fail, but we are done sending to imageLoadGoroutine.
+	d.pipeW.Close()
```

Comment:
No error checking? The original `d.writer.Close()` did do an error check, why this change?